### PR TITLE
Model Registry integration tests 

### PR DESCRIPTION
**Pull request by mparkhe**
## What changes are proposed in this pull request?

1. Added integration tests for all Model Registry APIs

2. Backend fixes in `SqlAlchemyStore`:
 - `last_updated_time` for registered model on `UpdateRegisteredModel`, `CreateModelVersion`, `DeleteModelVersion`, `UpdateModelVersion` (stage only)
 - `last_updated_time` from model versions on `UpdateModelVersion`
 - `CreateModelVersion` after `DeleteModelVersion` does not reuse version number in any case

## How is this patch tested?

Added new integration test suite

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [x] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [x] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/tracking/test_model_registry.py [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+
+    mv3 = mlflow_client.create_model_version(name, "another_path/to/model", "run_id_2")
+    assert mv3.version == 3
+    assert [mvd1] == mlflow_client.search_model_versions("source_path = 'path/to/model'")
+    assert [mvd1, mvd2] == mlflow_client.search_model_versions("run_id = 'run_id_1'")
+
+    assert "path/to/model" == mlflow_client.get_model_version_download_uri(name, 1)
+
+
+def test_update_create_model_version_flow(mlflow_client, backend_store_uri):
```
Comment (sueann) [inline tests/tracking/test_model_registry.py]: nit: `test_update_model_version_flow` ?

File: tests/tracking/test_model_registry.py [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+    assert_is_between(start_time_0, end_time_0, rmd2.creation_timestamp)
+    assert_is_between(start_time_1, end_time_1, rmd2.last_updated_timestamp)
+
+    mv2 = mlflow_client.create_model_version(name, "another_path/to/model", "run_id_1")
+    assert mv2.version == 2
+    assert mv2.registered_model.name == name
+    mv3 = mlflow_client.create_model_version(name, "a/b/c", "run_id_2")
+    assert mv3.version == 3
+    assert mv3.registered_model.name == name
+    model_versions_detailed = [rm.latest_versions
```
Comment (sueann) [inline tests/tracking/test_model_registry.py]: do we have any tests explicitly calling `client.get_latest_versions` ?

**Review by sueann — COMMENTED** [review]
looks mostly good to me - a few nits & questions inline. Thanks!

File: tests/tracking/test_model_registry.py (hunk 2 above)
Comment (mparkhe) [reply tests/tracking/test_model_registry.py]: added

File: tests/tracking/test_model_registry.py [hunk 3]
```diff
@@ -386,6 +299,40 @@ def test_update_create_model_version_flow(mlflow_client, backend_store_uri):
...
+
+    version_mvd_mapping = {}
+    for version, stage in version_stage_mapping.items():
+        mv = mlflow_client.create_model_version(name, "path/to/model", "run_id")
+        assert mv.version == version
+        if stage != "None":
+            mlflow_client.update_model_version(name, version, stage=stage)
+        mvd = mlflow_client.get_model_version_details(name, version)
+        assert mvd.current_stage == stage
+        version_stage_mapping[version] = mvd
```
Comment (sueann) [inline tests/tracking/test_model_registry.py]: do you need this? seems like we are unnecessarily modifying the map in flight?

**Review by sueann — APPROVED** [review]
lgtm - thanks for doing this! just one comment inline.

File: tests/tracking/test_model_registry.py (hunk 3 above)
Comment (mparkhe) [reply tests/tracking/test_model_registry.py]: you are right! Not needed anymore...

File: tests/__init__.py [hunk 4]
```diff
@@ -1,3 +1,3 @@
-from mlflow.utils.logging_utils import _configure_mlflow_loggers
+#from mlflow.utils.logging_utils import _configure_mlflow_loggers
```
Comment (mparkhe) [inline tests/__init__.py]: backed out this accidental change change in `7018d2473aacb95290028bf0f7546c706c0acb5a`

**Review by sueann — APPROVED** [review]
thank you for making tests/tracking lint-compatible!! i see that it needed some clean up as well, so it seems like a good idea anyway.

**Pull request merged**